File: app/src/main/java/castofo/com/co/nower/models/Promo.java
package castofo.com.co.nower.models;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

import io.realm.RealmObject;
import io.realm.RealmResults;
import io.realm.annotations.LinkingObjects;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Created by alejandrosanchezaristizabal on 9/4/17.
 */

public class Promo extends RealmObject {

  @PrimaryKey
  @Required
  private String id;
  private String name;
  private String description;
  private String terms;
  private Integer stock;
  private Float price;
  @SerializedName("start_date")
  private Date startDate;
  @SerializedName("end_date")
  private Date endDate;
  @LinkingObjects("promos")
  private final RealmResults<Branch> branches = null;

  public Promo() {
  }

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public String getTerms() {
    return terms;
  }

  public void setTerms(String terms) {
    this.terms = terms;
  }

  public Integer getStock() {
    return stock;
  }

  public void setStock(Integer stock) {
    this.stock = stock;
  }

  public Float getPrice() {
    return price;
  }

  public void setPrice(Float price) {
    this.price = price;
  }

  public Date getStartDate() {
    return startDate;
  }

  public void setStartDate(Date startDate) {
    this.startDate = startDate;
  }

  public Date getEndDate() {
    return endDate;
  }

  public void setEndDate(Date endDate) {
    this.endDate = endDate;
  }
}

File: app/src/main/java/castofo/com/co/nower/models/Promo.kt
package castofo.com.co.nower.models

import com.google.gson.annotations.SerializedName

import io.realm.RealmObject
import io.realm.RealmResults
import io.realm.annotations.LinkingObjects
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required

/**
 * Created by Santiago Vanegas on 12/9/17.
 */
open class Promo(
        @PrimaryKey
        @Required
        var id: String? = null,
        var name: String? = null,
        var description: String? = null,
        var terms: String? = null,
        var stock: Int = 0,
        var price: Float = 0.toFloat(),
        @SerializedName("start_date")
        var startDate: String? = null,
        @SerializedName("end_date")
        var endDate: String? = null,
        @LinkingObjects("promos")
        val branches: RealmResults<Branch>? = null
) : RealmObject()
